cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TranslatorSuite.scala

```diff
@@ -0,0 +1,434 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.cognitive.split1
+
+import com.microsoft.ml.spark.Secrets
+import com.microsoft.ml.spark.cognitive._
+import com.microsoft.ml.spark.core.test.base.{Flaky, TestBase}
+import com.microsoft.ml.spark.core.test.fuzzing.{TestObject, TransformerFuzzing}
+import org.apache.spark.ml.util.MLReadable
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions.{col, flatten}
+import org.scalactic.Equality
+
+trait TranslatorKey {
+  lazy val translatorKey: String = sys.env.getOrElse("TRANSLATOR_KEY", Secrets.TranslatorKey)
+
+  lazy val translatorName: String = sys.env.getOrElse("TRANSLATOR_NAME", Secrets.TranslatorName)
+}
+
+trait TranslatorUtils extends TestBase {
+
+  import spark.implicits._
+
+  lazy val textDf1: DataFrame = Seq(List("Hello, what is your name?")).toDF("text")
+
+  lazy val textDf2: DataFrame = Seq(List("Hello, what is your name?", "Bye")).toDF("text")
+
+  lazy val textDf3: DataFrame = Seq(List("This is bullshit.")).toDF("text")
+
+  lazy val textDf4: DataFrame = Seq(List("<div class=\"notranslate\">This will not be translated." +
+    "</div><div>This will be translated.</div>")).toDF("text")
+
+  lazy val textDf5: DataFrame = Seq(List("The word <mstrans:dictionary translation=wordomatic>word " +
+    "or phrase</mstrans:dictionary> is a dictionary entry.")).toDF("text")
+
+}
+
+class TranslateSuite extends TransformerFuzzing[Translate]
+  with TranslatorKey with Flaky with TranslatorUtils {
+
+  lazy val translate: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setToLanguage(Seq("zh-Hans"))
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  lazy val translate2: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setToLanguage(Seq("zh-Hans"))
+    .setToScript("Latn")
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  lazy val translate3: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage(Seq("fr"))
+    .setIncludeAlignment(true)
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  lazy val translate4: Translate = new Translate()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage(Seq("fr"))
+    .setIncludeSentenceLength(true)
+    .setTextCol("text")
+    .setOutputCol("translation")
+    .setConcurrency(5)
+
+  test("Translate multiple pieces of text with language autodetection") {
```

Comment:
all of these tests have a similiar structure, might want to factor this structure out so that the tests are smaller to write. I know it's just test code and who cares, but it will make your life easier when you need to update I promise

Reply:
Done.